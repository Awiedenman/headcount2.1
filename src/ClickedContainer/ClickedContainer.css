.clicked-container {
  display: flex;
  justify-content: center;
  height: 340px;
}

.clicked-container div {
  /* display: inline-block; */

}