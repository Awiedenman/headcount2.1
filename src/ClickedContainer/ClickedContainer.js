import React from 'react';
import { Card } from '../Card/Card';
import PropTypes from 'prop-types';
import './ClickedContainer.css';
import { ComparisonCard } from '../ComparisonCard/ComparisonCard';

export const ClickedContainer = ( { clickedCard1, clickedCard2, addClickedCard, compareDistrictAverages, findAverage }) => {
  //remove filter part and just pass in two districts to component
  // const renderClickedCards = allDistricts.filter(( district ) => {
  //   return district.clicked;
  // });
  let clickedCards;
  if(clickedCard1 && clickedCard2) {
    clickedCards = [clickedCard1, clickedCard2]
  } else if(clickedCard1) {
    clickedCards = [clickedCard1]
  } else if(clickedCard2) {
    clickedCards = [clickedCard2]
  } else {
    clickedCards = [];
  }
  const compareCards = clickedCards.map(( clickedDistrict, index) => {
    return ( 
      <Card 
        key={index}
        location={clickedDistrict.location}
        stats={clickedDistrict.stats}
        addClickedCard={addClickedCard}
        clicked={ clickedDistrict.clicked }
      />
    );
  });

  return (
    <div className='clicked-container'>
      <ComparisonCard 
        clickedCards={ clickedCards }
        compareDistrictAverages={ compareDistrictAverages }
        findAverage={findAverage}
      />
      { compareCards }
    </div>
  );
};

ClickedContainer.propTypes = {
  // allDistricts: PropTypes.array,
  addClickedCard: PropTypes.func,
  findAverage: PropTypes.func,
  compareDistrictAverages: PropTypes.func
};