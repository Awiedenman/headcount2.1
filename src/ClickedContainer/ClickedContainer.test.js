import React from 'react';
import { shallow } from 'enzyme';
import { ClickedContainer } from './ClickedContainer';

describe('ClickedContainer', ()=> {
  const mockAllDistricts = [];
  const mockAddClickedCard = jest.fn();
  const mockCompareDistrictAverages = jest.fn();
  const mockFindAverage = jest.fn();
  it('should match snapshot upon render', () => {
    const wrapper = shallow(<ClickedContainer
                              allDistricts={ mockAllDistricts }
                              addClickedCard={ mockAddClickedCard }
                              compareDistrictAverages={ mockCompareDistrictAverages }
                              findAverage={ mockFindAverage }
                            />);
    expect(wrapper).toMatchSnapshot();
  })





})