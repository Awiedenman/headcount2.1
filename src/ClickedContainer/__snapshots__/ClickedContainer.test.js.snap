// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickedContainer should match snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addClickedCard={[Function]}
    allDistricts={Array []}
    compareDistrictAverages={[Function]}
    findAverage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <Unknown
          clickedCards={Array []}
          compareDistrictAverages={[Function]}
          findAverage={[Function]}
/>,
        undefined,
      ],
      "className": "clicked-container",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clickedCards": Array [],
          "compareDistrictAverages": [Function],
          "findAverage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <Unknown
            clickedCards={Array []}
            compareDistrictAverages={[Function]}
            findAverage={[Function]}
/>,
          undefined,
        ],
        "className": "clicked-container",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clickedCards": Array [],
            "compareDistrictAverages": [Function],
            "findAverage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
